test.cmd

```diff
@@ -0,0 +1,24 @@
+@echo off
+setlocal
+cd "%~dp0"
+chcp 1252 > nul
+if not exist tools\NUnit.Runners (
+    for %%i in (NuGet.exe) do set nuget=%%~dpnx$PATH:i
+    if "%nuget%"=="" goto :nonuget
```

Comment:
This check here causes `test.cmd` to _always_ fail with the following output irrespective of it `NuGet.exe` is in the path or not:

```
NuGet executable not found in PATH
For more on NuGet, see http://nuget.codeplex.com
```

The reason is that the outer `if` along with its block is expanded _before_ being evaluated so `if "%nuget%"=="" goto :nonuget` becomes `if ""=="" goto :nonuget` at that point and will always be true. I would recommend using a simpler approach by refactoring the block as follows (so that each line is expanded as its evaluated):

```cmd
if exist tools\NUnit.Runners goto :test
for %%i in (NuGet.exe) do set nuget=%%~dpnx$PATH:i
if "%nuget%"=="" goto :nonuget
md tools
nuget.exe install NUnit.Runners -Version 2.6.4 -OutputDirectory tools
move .\tools\NUnit.Runners.2.6.4 .\tools\NUnit.Runners
if not %errorlevel%==0 exit /b %errorlevel%
:test
```